prow/plugins/override/override.go

```diff
@@ -135,6 +155,70 @@ func authorized(gc githubClient, log *logrus.Entry, org, repo, user string) bool
 	return ok
 }
 
+func authorizedTopLevelOwner(oc ownersClient, allowTopLevelOwners bool, log *logrus.Entry, org, repo, user string, pr *github.PullRequest) bool {
+	if allowTopLevelOwners {
+		owners, err := oc.LoadRepoOwners(org, repo, pr.Base.Ref)
+		if err != nil {
+			log.WithError(err).Warnf("cannot determine whether %s is a top level owner of %s/%s", user, org, repo)
+			return false
+		}
+
+		if owners.TopLevelApprovers().Has(user) {
+			return true
+		}
+	}
+	return false
+}
+
+func validateGitHubTeamSlugs(teamSlugs map[string][]string, org, repo string, githubTeams []github.Team) error {
+	slugs := sets.NewString()
+	for _, team := range githubTeams {
+		slugs.Insert(team.Slug)
+	}
+
+	repoRef := fmt.Sprintf("%s/%s", org, repo)
+	invalidSlugs := sets.NewString()
+	for _, slug := range teamSlugs[repoRef] {
+		if !slugs.Has(slug) {
+			invalidSlugs.Insert(slug)
+		}
+	}
+	if invalidSlugs.Len() > 0 {
+		return fmt.Errorf("invalid team slug(s): %s", strings.Join(invalidSlugs.List(), ","))
+	}
+	return nil
+}
+
+func authorizedGitHubTeamMember(gc githubClient, log *logrus.Entry, teamSlugs map[string][]string, org, repo, user string) bool {
+	repoRef := fmt.Sprintf("%s/%s", org, repo)
+	teams, err := gc.ListTeams(org)
```

Comment:
Where do we filter teams by their role on the repo? You might want [this](https://developer.github.com/v3/repos/#list-teams) endpoint instead.

Reply:
Thank you for the review. This was based on my interpretation of the comments on the issue that this was intended to allow override regardless of GitHub's teams' access to a repo, i.e. not necessarily follow GitHub teams' permissions on repositories. Could you please advise if only teams that manage a repo should be allowed to override?

Reply:
Ah, you are correct! My apologies :)